crypto/evp/evp_enc.c

```diff
@@ -336,6 +354,18 @@ int EVP_EncryptUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl,
             return 0;
         }
     }
+
+    /*
+     * Aliasing in and out in this way is counter intuitive and only supported
+     * for legacy applications.
+     * Don't rely on this to stay the same in the next major version.
+     */
+    if (is_any_overlapping(out, inl + (bl - 1), in, inl)
+            && (PTRDIFF_T)in != (PTRDIFF_T)(out + ctx->buf_len)) {
```

Comment:
Two things [for now?]. Shouldn't it be `cmpl` instead of `inl`? Secondly I'd like to suggest to flip the terms of the expression, i.e. compare `in`  and `out + ctx->buf_len` first. It's more readable and takes less computations.

On related note. Looking few lines down we see `i = ctx->buf_len`. Since we cache the value, we can as well start earlier...

Reply:
The code below assumes inl is in bytes, and out is != NULL.
So if EVP_CIPH_FLAG_LENGTH_BITS is set, the cipher does
very likey not support bit lengths, nor AAD.

Reply:
> The code below assumes inl is in bytes

In such case wouldn't it be appropriate to add a commentary that says "past this point inl is bytes"?

Reply:
Yes, good point. I will update the comment accordingly.